analysis/benchmark_results.py

```diff
@@ -53,16 +56,37 @@ def _benchmark_df(self):
         exp_df = self._experiment_df
         return exp_df[exp_df.benchmark == self.name]
 
+    @property
+    @functools.lru_cache()
+    def _fuzzer_names(self):
+        """Names of valid fuzzers."""
+        return self._benchmark_df.fuzzer.unique()
+
     @property
     @functools.lru_cache()
     def _benchmark_snapshot_df(self):
         return data_utils.get_benchmark_snapshot(self._benchmark_df)
 
     @property
     @functools.lru_cache()
-    def fuzzers(self):
-        """Fuzzers with valid trials on this benchmark."""
-        return self._benchmark_df.fuzzer.unique()
+    def _benchmark_coverage_dict(self):
+        """Covered regions of each fuzzer on this benchmark."""
```

Comment:
"region" is a term that is very specific to clang coverage. I'd prefer we didn't use it internally and we definitely should not use it in things that are displayed to users.
I also realized another problem with using regions. Currently we store "edges_covered" in the db for each snapshot.
However, I'm not sure that each region in clang-cov is a unique edge. In sancov this was the case because the only instrumented blocks were dummy blocks that were added to map uniquely to a single edge.
@Dor1s @morehouse any thoughts on this? Do you know if region based coverage is more akin to blocks? Max: I think when we compared clang cov and sancov they generally said the same thing

Reply:
I'm not sure the exact definition of region in clang cov, but from [the docs](https://clang.llvm.org/docs/SourceBasedCodeCoverage.html):

```
Region coverage is the percentage of code regions which have been executed at least once.
A code region may span multiple lines (e.g in a large function body with no control flow).
However, it’s also possible for a single line to contain multiple code regions
(e.g in “return x || y && z”).
```

Sounds similar to "edge coverage", or block coverage with critical edges split.

Reply:
+1 on not worrying about these definitions too much. i think it's fine to use "region" in the code that specifically deals with "clang coverage", and it's fine to keep "edge coverage" elsewhere. these terms (like "edge") of course mean slightly different things on different abstraction levels (ie. source, IR, machine code), but i think the bottom line is that clang coverage gives pretty much the most fine grained src coverage info (ie. more fine grained than just  block/line/statement/branch/etc). i guess that's why they call it "region", because it captures all these things, including boolean expression like “return x || y && z”

Reply:
OK, but it should not be "region" in the report we output.

Reply:
Yes, the report itself should def not use "region".

Reply:
I'll have to investigate later but it sounds more like block than edge to me. If I think of the example explaining how sancov instruments edges:
https://clang.llvm.org/docs/SanitizerCoverage.html#edge-coverage
I would imagine that clang cov cannot tell that edge A-C was taken, since this isn't useful information for displaying in a coverage report.

Reply:
Ah, my bad, i wrongly assumed/remebered that source based coverage does branch coverage too, just like gcov (ie. when compiling with `--coverage -fprofile-arcs`) but it actually doesn't. So yes, in case of that example, it wouldn't able to tell if A-C was taken when A-B-C was taken (only gcov). Considering the precision of the coverage information, I don't think this matters too much (the example is an edge case), but "edge coverage" might indeed not be a good name. I'd personally just say "coverage" in plots/tables/etc, and explain somewhere in the report that we're specifically using https://clang.llvm.org/docs/SourceBasedCodeCoverage.html. Wdyt?